mt-kahypar/partition/refinement/flows/problem_construction.h

```diff
@@ -111,7 +111,14 @@ class ProblemConstruction {
     _context(context),
     _scaling(1.0 + _context.refinement.flows.alpha *
       std::min(0.05, _context.partition.epsilon)),
-    _local_bfs(num_hypernodes, num_hyperedges, context.partition.k) { }
+    _local_bfs(
+      // If the number of blocks changes, BFSData needs to be initialized
+      // differently. Thus we use a lambda that reads the current number of
+      // blocks from the context.
+      [num_hypernodes, num_hyperedges, &context] {
```

Comment:
This is dangerous as it captures the variables in the constructor which may be not available at the time this function is called. It would be better if there is a local function that access members of the class.

Reply:
For this reason num_hypernodes / num_hyperedges is captured by copy, so it shouldn't be a problem. But I can change it anyway